core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -65,6 +65,12 @@ private[spark] class Executor(
 
   logInfo(s"Starting executor ID $executorId on host $executorHostname")
 
+  @volatile private var executorShutdown = false
+  ShutdownHookManager.addShutdownHook(
+    () => if (!executorShutdown) {
```

Comment:
You don't need to check this here.

Reply:
If i don't check it here then stop() method will be called twice in case of graceful shutdown.

Reply:
No, because stop() checks it.

Reply:
Please review it.